src/wallet3/pending_transaction.hpp

```diff
@@ -35,18 +35,40 @@ namespace wallet
 
     std::vector<Output> chosenOutputs;
 
+    bool blink = true;
+
+    int64_t fee = 0;
+    uint64_t fee_per_byte = FEE_PER_BYTE_V13;
+    uint64_t fee_per_output = FEE_PER_OUTPUT_V18;
+    int8_t mixin_count = CRYPTONOTE_DEFAULT_TX_MIXIN;
+    size_t extra_size() const {return 0;};
+
     PendingTransaction() = default;
 
     PendingTransaction(const std::vector<TransactionRecipient>& new_recipients);
 
+    void
+    SetFeePerByte(int64_t per_byte) { fee_per_byte = per_byte; };
```

Comment:
why does this function exist?

Reply:
Yeah so the thought process behind the setter function for `fee_per_byte` member was to make it private eventually. Might be better just doing a public member and modifying it directly.

However the way the code is moving this `fee_per_byte` (Also `fee_per_output`) around feels a little off and wouldn't mind discussing it further.

The `PendingTransaction` class has a function `GetFee()` so you can call and know what the fee would be using the PendingTransactions current state. The `transaction_constructor` sets the fee parameters but it also defaults to some reasonable constants.

And the Transaction Constructor has something similar when it gets constructed. It immediately calls a `daemon_comms` function to get the fee parameters. The Daemon Comms class having a periodic rpc call with the daemon to keep these parameters up to date.

And all 3 - Pending Transaction, Transaction Constructor, Daemon Comms, have defaults pulled from `oxen_config` constants which have getters and setters to use. Which im not sure if there is a better way

Reply:
>Might be better just doing a public member and modifying it directly.

i think this is the only correct solution here, setter functions are an anti pattern.